### ISSUE TITLE
Answering questions from Q&A corpus

### ISSUE THREAD

**Issue by jrthom18** [jcjohnson/torch-rnn#154]
Has anybody had any success with a Q&A corpus using this implementation? I trained on a ~6MB corpus with questions and their corresponding answers on alternating lines in the source text file from a public Stack Exchange dump. After training for about 24 hours I achieved a training loss of ~1.0 and val loss of 1.12. Then I tried to test it by seeding the sampling script with a question for -start_text to see if it would produce the answer (also -sample flag set to 0 to get argmax at each step). The generated text was good coherent English and of course relevant to the topics trained on but just not anywhere close to answering the actual question. Any ideas?

**Comment by jcjohnson** [comment]
This isn't really the right tool for question-answering. Rather than training a single RNN on an endless stream of interleaved questions and answers, it is much more common to treat Q&A as a supervised learning problem, where you have independent input/output pairs (question, answer). You can then train separate encoder and decoder RNNs, where the encoder reads the question and outputs a vector and the decoder tries to reconstruct the answer from the encoded question.

**Comment by jrthom18** [comment]
@jcjohnson ah ok that makes sense now. Any good implementations you know of that use encoder and decoder RNNs? I could modify the Tensorflow seq2seq tutorial code to do it. I've been playing around with that one which tokenizes at the word level out of the box.

**Comment by jcjohnson** [comment]
I'm not sure if there is anything you can just pick up and use. Modifying the TensorFlow seq2seq example would probably work; you could also take pieces from torch-rnn if you wanted to do it in Torch.

**Comment by jrthom18** [comment]
@jcjohnson Thanks for the help!
